gensim/similarities/termsim.py

```diff
@@ -156,32 +140,288 @@ def most_similar(self, t1, topn=10):
                     yield (t2, similarity**self.exponent)
 
 
+def _shortest_uint_dtype(max_value):
+    """Get the shortest unsingned integer data-type required for representing values up to a given
+    maximum value.
+
+    Returns the shortest unsingned integer data-type required for representing values up to a given
+    maximum value.
+
+    Parameters
+    ----------
+    max_value : int
+        The maximum value we wish to represent.
+
+    Returns
+    -------
+    data-type
+        The shortest unsigned integer data-type required for representing values up to a given
+        maximum value.
+    """
+    if max_value < 2**8:
+        return np.uint8
+    elif max_value < 2**16:
+        return np.uint16
+    elif max_value < 2**32:
+        return np.uint32
+    return np.uint64
+
+
+def _create_source(index, dictionary, tfidf, symmetric, dominant, nonzero_limit, dtype):
+    """Build a sparse term similarity matrix using a term similarity index.
+
+    Returns
+    -------
+    matrix : :class:`scipy.sparse.coo_matrix`
+        The sparse term similarity matrix.
+
+    """
+    assert isinstance(index, TermSimilarityIndex)
+    assert dictionary is not None
+    matrix_order = len(dictionary)
+
+    if matrix_order == 0:
+        raise ValueError('The provided dictionary is empty')
```

Comment:
That's more a statement of fact than an error :) What's the actual problem?

Ideally, we want to show user the error (what's wrong), plus also suggest a step to overcome the problem (not sure that applies here, just in general).

Reply:
I am not sure we have enough information here to say anything more meaningful. We may want to prohibit the creation of empty dictionaries or at least raise a warning if we worry that this may break existing code bases.

Reply:
Even then, a message like `Dictionary in SparseTermSimilarity must not be empty` (or similar) is better.

Reply:
I updated the message in 8584880.